### ISSUE TITLE
Effects viability needs better timing, especially for Sitreps

### ISSUE THREAD

**Issue by MatGB** [freeorion/freeorion#1623]
### Environment
* **FreeOrion Version**: f0d71bd
* **Operating System**: Linux
  * Compiled from source

### Description
Multiple times Effects either do or don't work as intended or are announced as having worked differently to the way they should as a result of the way scope and activation are timed.

I know this is a complex issue but, at some point, it needs fixing.

Examples:
[A building completing and effects priority](https://github.com/freeorion/freeorion/pull/1614#issuecomment-312554247)
Drydock repairs vs damage control techs, drydocks are timed to occur last, but you get a 'partial repair' sitrep for a ship that's fully repaired by other effects
Drydock repair and stargates: this may have been closed but you used to get a sitrep saying drydock a had repaired a ship that then went through a stargate, if location b didn't have a drydock no repair took place but it was announced as complete (think this was fixed inadvertently but it's an example)
If a colony building completes and colonises a planet in a turn when a monstrous bioweapon is in system, the bioweapon will depopulate the new colony but no sitrep will happen (the colony wasn't there when the weapon scope was calculated) (see pic: the Mu Ursh world has lost population then gained some from growth, it should be above 3 as I have Lifecycle)

The biggest timing issue is for sitreps: they should be accurate and describe what actually happened, not what might happen if nothing else affects things. But there can also be other timing interactions that create issues such as in the link and, for example, the Experimentor outpost can create monsters after it's captured but before it's destroyed, giving the player a fleet of nasty beasts.

### Expected Result
Sitreps should describe what actually happened, other effects should or should not trigger based on current situation not what was first assessed in scope.
![screenshot_20170703_173925](https://user-images.githubusercontent.com/11533206/27801804-ea5beb36-6017-11e7-82c6-0a9217fd7a23.png)

**Comment by MatGB** [comment]
I've tagged this bug and refactoring, as well as both infrastructure and UI as it crosses over, someone more understanding of the backend might want to tag it more accurately.

**Comment by Vezzra** [comment]
Huh, that's difficult to classify correctly, as this is a quite broad issue. I've adjusted the labels to something I think is more appropriate. ["component:infrastructure"](https://github.com/freeorion/freeorion/blob/master/.github/labels.md#label-componentinfrastructure) in particular is for something completely different:

> The Issue/PR deals with non-game related infrastructure that aids development. For example this includes issues with the project homepage, used external services like continuous integration or hosting of release binaries.

**Comment by geoffthemedio** [comment]
internal probably

**Comment by Vezzra** [comment]
Makes sense... it's not really an issue with the UI itself, and also not something wrong with the content scripts, but rather an issue with the underlying engine.
